operators/src/processing/interpolation/mod.rs

```diff
@@ -0,0 +1,824 @@
+) -> impl Future<Output = Result<InterpolationAccu<T, I>>> {
+    // create an accumulator as a single tile that fits all the input tiles
+    crate::util::spawn_blocking(move || {
+        let tiling = tiling_specification.strategy(
+            query_rect.spatial_resolution.x,
+            -query_rect.spatial_resolution.y,
+        );
+
+        // TODO: enlarge query_rect to be sure to have all the neighbor pixels for the edges
+
+        let origin_coordinate = tiling
+            .tile_information_iterator(query_rect.spatial_bounds)
+            .next()
+            .expect("a query contains at least one tile")
+            .spatial_partition()
+            .upper_left();
+
+        let geo_transform = GeoTransform::new(
+            origin_coordinate,
+            query_rect.spatial_resolution.x,
+            -query_rect.spatial_resolution.y,
+        );
+
+        let bbox = tiling.tile_grid_box(query_rect.spatial_bounds);
+
+        let shape = [
+            bbox.axis_size_y() * tiling.tile_size_in_pixels.axis_size_y(),
+            bbox.axis_size_x() * tiling.tile_size_in_pixels.axis_size_x(),
+        ];
+
+        // create a non-aligned (w.r.t. the tiling specification) grid by setting the origin to the top-left of the tile and the tile-index to [0, 0]
+        let grid = Grid2D::new(
+            shape.into(),
+            vec![no_data_value; shape[0] * shape[1]],
+            Some(no_data_value),
+        )
+        .expect("grid creation must not fail");
+
+        let input_tile = RasterTile2D::new(
+            query_rect.time_interval,
+            [0, 0].into(),
+            geo_transform,
+            GridOrEmpty::Grid(grid),
+        );
+
+        let output_tile = RasterTile2D::new_with_tile_info(
+            query_rect.time_interval,
+            tile_info,
+            GridOrEmpty::Empty(EmptyGrid::new(
+                tiling_specification.tile_size_in_pixels,
+                no_data_value,
+            )),
+        );
+
+        InterpolationAccu::new(input_tile, output_tile, pool)
+    })
+    .map_err(From::from)
+}
+
+pub fn fold_future<T, I>(
+    accu: InterpolationAccu<T, I>,
+    tile: RasterTile2D<T>,
+) -> impl Future<Output = Result<InterpolationAccu<T, I>>>
+where
+    T: Pixel,
+    I: InterpolationAlgorithm<T>,
+{
+    crate::util::spawn_blocking(|| fold_impl(accu, tile)).then(|x| async move {
+        match x {
+            Ok(r) => r,
+            Err(e) => Err(e.into()),
+        }
+    })
+}
+
+pub fn fold_impl<T, I>(
+    mut accu: InterpolationAccu<T, I>,
+    tile: RasterTile2D<T>,
+) -> Result<InterpolationAccu<T, I>>
+where
+    T: Pixel,
+    I: InterpolationAlgorithm<T>,
+{
+    // get the time now because it is not known when the accu was created
+    accu.output_tile.time = tile.time;
+
+    // copy all input tiles into the accu to have all data for interpolation
+    let mut accu_input_tile = accu.input_tile.into_materialized_tile();
+    accu_input_tile.blit(tile)?;
```

Comment:
i see the benefit of using blit here. However my suggestion would be to use a Buffer of 4 / 9 tiles and implement pixel access on that buffer. Then we can avoid the copy operation each blit requres.

Reply:
this would avoid memcopy but destroy data locality as the rows will be split up so I don't think its 100% clear that this is more efficient so I wouldn't prematurely optimize this.